core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -74,7 +74,17 @@ object StaticFile {
       }
     })
 
-  def calcETag[F[_]: Sync]: File => F[String] = f => Sync[F].delay(f.length().toString)
+  def calcETag[F[_]: Sync]: File => F[String] =
+    f =>
+      Sync[F]
+        .delay(f.isFile)
+        .flatMap { isFile =>
+          if (isFile) {
```

Comment:
We could probably micro-optimize this into a single delay rather than a flatMap.  It's all local.

Reply:
Single flatMap as in call unsafeRun on calcETag or not have F on it? It was without F before because it was only ever called from inside a delay block.

Reply:
No. Instead of `.delay(...).flatMap(...)` just do `.delay(...)`

Reply:
oops, wrong place 👅